File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/SchedulingInfs/Commands/SchedulingInfValidCommand.js
SIE.defineCommand('SIE.Web.MES.TaskManagement.SchedulingInfs.Commands.SchedulingInfValidCommand', {
    meta: { text: "排程校验", group: "business", iconCls: "icon-TextQuality icon-blue" },

    canExecute: function (view) {
        if (view && view.getSelection() && view.getSelection().length > 0 && view.getSelection().all(p => p.getIsCheck() == false || p.getIsCheck() == null))
            return true;
        return false;
    },

    execute: function (view) {
        var ids = view.getSelectionIds();
        view.execute({
            data: ids,
            success: function (res) {
                if (res.Success) {
                    view.reloadData();
                }
            }
        });
    }

});

File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/SchedulingInfs/Commands/SchedulingInfValidCommand.cs
using SIE.MES.TaskManagement.SchedulingInfs;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIE.Web.MES.TaskManagement.SchedulingInfs.Commands
{
    /// <summary>
    /// 排程校验
    /// </summary>
    public class SchedulingInfValidCommand : ViewCommand<List<double>>
    {
        protected override object Excute(List<double> args, string scope)
        {
            RT.Service.Resolve<SchedulingInfController>().ValidSchedulingInf(args);
            return true;
        }
    }
}
